### ISSUE TITLE
[vscode] workspace registerFileSystemProvider options can explain why FS is readonly with 1.86 

### ISSUE THREAD

**Issue by rschnekenbu** [eclipse-theia/theia#13353]
With vscode 1.86, workspace.registerFileSystemProvider() readonly option can now be a markdown string to explain why the file system is readonly. 

see https://github.com/microsoft/vscode/blob/e5447598324c57ac814c150148e7e4c830e0f9ee/src/vscode-dts/vscode.d.ts#L13663

**Comment by rschnekenbu** [comment]
On looking to that issue, I stumbled upon #8167. I do agree with the issue of relative / absolute uri transformation which is done even for a folder using another file system provider. Workspace file in VS Code keeps the URI as it is, where we can only have uris with the 'file:/' scheme in ours.

**Comment by msujew** [comment]
@rschnekenbu You don't need to implement this feature from scratch, see https://github.com/eclipse-theia/theia/pull/13403. It should lay most of the groundwork for the this issue.

**Comment by rschnekenbu** [comment]
Indeed, I was implementing similar parts. I rebased my work on your changes. 
I can then also participate to your PR easily.
